test/e2e/edge-compiler-can-import-blob-assets/app/pages/api/edge.js

```diff
@@ -0,0 +1,33 @@
+export const config = { runtime: 'experimental-edge' }
+
+/**
+ * @param {import('next/server').NextRequest} req
+ */
+export default async (req) => {
+  const handlerName = req.nextUrl.searchParams.get('handler')
+  const handler = handlers.get(handlerName) || defaultHandler
+  return handler()
+}
+
+/**
+ * @type {Map<string, () => Promise<Response>>}
+ */
+const handlers = new Map([
+  [
+    'text-file',
+    async () => {
+      const url = new URL('../../src/text-file.txt', import.meta.url)
+      return fetch(url)
+    },
+  ],
+  [
+    'image-file',
+    async () => {
+      const url = new URL('../../src/vercel.png', import.meta.url)
+      return fetch(url)
+    },
+  ],
```

Comment:
Maybe add an fetch to an external asset too, to make sure that still works as expected.

Reply:
wdym external asset? `fetch("https://raw.githubusercontent.com/vercel/next.js/canary/test/.stats-app/pages/nextjs.png")` or `fetch(new URL("https://raw.githubusercontent.com/vercel/next.js/canary/test/.stats-app/pages/nextjs.png"))` or is there a custom syntax for external assets? 🤔

Reply:
yep that one. To ensure the overriding fetch function still works correctly for non-blob urls